src/Microsoft.OData.Edm/Csdl/Parsing/JsonParserContext.cs

```diff
@@ -0,0 +1,227 @@
+        /// Gets the parsing source.
+        /// </summary>
+        public string Source { get; }
+
+        /// <summary>
+        /// Gets the CSDL-JSON reader settings.
+        /// </summary>
+        public CsdlJsonReaderSettings Settings { get; }
+
+        #region JsonPath
+        /// <summary>
+        /// Enter a JSON property scope.
+        /// </summary>
+        /// <param name="propertyName">The property name.</param>
+        public void EnterScope(string propertyName)
+        {
+            _nodes.Push(propertyName);
+            _dirty = true;
+        }
+
+        /// <summary>
+        /// Leave a JSON property/index scope.
+        /// </summary>
+        public void LeaveScope()
+        {
+            _nodes.Pop();
+            _dirty = true;
+        }
+
+        /// <summary>
+        /// Leave a JSON property scope.
+        /// </summary>
+        /// <param name="propertyName">The property name.</param>
+        public void LeaveScope(string propertyName)
+        {
+            Debug.Assert(_nodes.Count > 0);
+            Debug.Assert(_nodes.Peek().ToString() == propertyName);
+            _nodes.Pop();
+            _dirty = true;
+        }
+
+        /// <summary>
+        /// Entry an array index scope
+        /// </summary>
+        /// <param name="index">The array index.</param>
+        public void EnterScope(int index)
+        {
+            _nodes.Push(index);
+            _dirty = true;
+        }
+
+        /// <summary>
+        /// Leave a JSON array index scope.
+        /// </summary>
+        /// <param name="index">The array index.</param>
+        public void LeaveScope(int index)
+        {
+            Debug.Assert(_nodes.Count > 0);
+            Debug.Assert((int)_nodes.Peek() == index);
+            _nodes.Pop();
+            _dirty = true;
+        }
+
+        /// <summary>
+        /// Gets current JSON path string
+        /// </summary>
+        public string Path => BuildJsonPath();
+
+        /// <summary>
+        /// Gets current CsdlLocation.
+        /// </summary>
+        /// <returns>The built CsdlLocation.</returns>
+        public CsdlLocation Location()
+        {
+            return new CsdlLocation(BuildJsonPath());
+        }
+        #endregion
+
+        #region EdmError
+
+        /// <summary>
+        /// Gets all parsing errors
+        /// </summary>
+        public List<EdmError> Errors { get; }
+
+        /// <summary>
+        /// Reports a certain parsing error.
+        /// </summary>
+        /// <param name="errorCode">The Edm Error Code.</param>
+        /// <param name="errorMessage">The Error message.</param>
+        public void ReportError(EdmErrorCode errorCode, string errorMessage)
+        {
+            Errors.Add(new EdmError(Location(), errorCode, errorMessage));
+        }
+
+        /// <summary>
+        /// Check parsing status.
+        /// </summary>
+        /// <returns>True/false</returns>
+        public bool IsSuccessed()

```

Comment:
>IsSuccessed [](start = 20, length = 11)

Should be IsSuccess or Succeeded (or IsSucceeded)

Reply:
IsSucceeded

---
In reply to: [429398447](https://github.com/OData/odata.net/pull/1775#discussion_r429398447) [](ancestors = 429398447)